File: src/main/kotlin/com/jova/domain/auth/dto/response/TokenResponse.java
package com.jova.domain.auth.dto.response;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class TokenResponse {

    String accessToken;
    String refreshToken;
    String grantType;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    LocalDateTime accessTokenExpiration;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    LocalDateTime refreshTokenExpiration;
}

File: src/main/kotlin/com/jova/domain/auth/dto/response/TokenResponse.kt
package com.jova.domain.auth.dto.response

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime

data class TokenResponse(
    val accessToken: String,
    val refreshToken: String,
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss") val accessTokenExpiration: LocalDateTime,
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss") val refreshTokenExpiration: LocalDateTime
)
